lib/Map/RegionProvider.js

```diff
@@ -116,6 +117,17 @@ var RegionProvider = function(regionType, properties) {
      * @type {Object}
      */
     this._idIndex = {};
+
+    /**
+     * The URL of a pre-generated JSON file containing just a long list of IDs for a given
+     * layer attribute, in the order of ascending feature IDs (fids). If defined, it will
+     * be used in preference to requesting those attributes from the WFS server.
+     * @type {String}
+     */
+    this.regionIdList = properties.regionIdList;
```

Comment:
Interesting points. I had thought that the JSON spec required the outer most object to be an object, not an array, but it looks like it doesn't. It's still sort of frowned on in API design though - I prefer passing an object that has a tiny bit of context at least.

Personally I think it's ok to describe an object which consists of two strings and an array of thousands of values as "a list". It really is just a list with a tiny bit of metadata. Otherwise you'd have to call virtually everything an object - even a javascript `Array` is an object, right? :)

In any case, I'm not really proposing that the format of this JSON file be a public interface - it's a private format that is used in this one place, and by the script that generates it. 

>So, what would you say to renaming values to ids

This is really tricky. Depending on the exact context, you can call these things "values", "ids", "codes", "attributes" etc etc. I'm a bit uncomfortable calling them "ids" because for certain layers and attributes, they're English text strings like "Baw Baw (S)", which don't really have the character of "IDs" to me. At the time they're generated by the script, they're (agnostically) the value of a particular attribute on a particular layer. Later on, our code *treats* them as IDs, possibly too optimistically (as with ambiguous names like "Camperdown (C)").

> and renaming this to `this.regionIdsUrl`

Technically, it's not a URL, because it can just be a relative path, `data/regionids/foo.json`. I think `regionIdsFile` would be more accurate.

(Ha, I think I out-nitpicked your nitpick :))